mediapipe/tasks/python/test/vision/image_segmenter_test.py

```diff
@@ -280,20 +275,49 @@ def test_segment_for_video_with_out_of_order_timestamp(self):
           ValueError, r'Input timestamp must be monotonically increasing'):
         segmenter.segment_for_video(self.test_image, 0)
 
-  def test_segment_for_video(self):
+  def test_segment_for_video_in_category_mask_mode(self):
     options = _ImageSegmenterOptions(
         base_options=_BaseOptions(model_asset_path=self.model_path),
-        output_type=_OutputType.CATEGORY_MASK,
+        output_category_mask=True,
         running_mode=_RUNNING_MODE.VIDEO)
     with _ImageSegmenter.create_from_options(options) as segmenter:
       for timestamp in range(0, 300, 30):
-        category_masks = segmenter.segment_for_video(self.test_image, timestamp)
-        self.assertLen(category_masks, 1)
+        segmentation_result = segmenter.segment_for_video(
+            self.test_image, timestamp)
+        category_mask = segmentation_result.category_mask
         self.assertTrue(
-            _similar_to_uint8_mask(category_masks[0], self.test_seg_image),
+            _similar_to_uint8_mask(category_mask, self.test_seg_image),
             f'Number of pixels in the candidate mask differing from that of the '
             f'ground truth mask exceeds {_MASK_SIMILARITY_THRESHOLD}.')
 
+  def test_segment_for_video_in_confidence_mask_mode(self):
+    # Load the cat image.
+    test_image = _Image.create_from_file(
+      test_utils.get_test_data_path(
+        os.path.join(_TEST_DATA_DIR, _CAT_IMAGE)))
+
+    options = _ImageSegmenterOptions(
+      base_options=_BaseOptions(model_asset_path=self.model_path),
+      running_mode=_RUNNING_MODE.VIDEO)
```

Comment:
please explicitly set output_category_mask=False, output_confidence_masks=True.
and for category mode test, set output_category_mask=True, output_confidence_masks=False.
ditto for live stream mode.

Reply:
I happened to not set those flags because the C++ API happened to use the following defaults:

https://github.com/google/mediapipe/blob/78499b0bd0e4304bfb6eaa6029754681367673f9/mediapipe/tasks/cc/vision/image_segmenter/image_segmenter.h#L57
(The Python API uses the same defaults) 

But yes, I believe it's better to explicitly state the mode.